neptune-core/src/application/rpc/server.rs

```diff
@@ -4163,9 +4179,96 @@ impl RPC for NeptuneRPCServer {
             .map(|tx| &tx.kernel)
             .cloned())
     }
+
+    // Crashes if `tx_ix` out of its bound. https://github.com/Neptune-Crypto/neptune-core/issues/816
+    async fn prove_transfer(
+        self,
+        context: ::tarpc::context::Context,
+        token: auth::Token,
+        tx_ix: u64,
+        utxo_ix: usize,
+        block: Digest,
+    ) -> RpcResult<(Claim, NeptuneProof)> {
+        if let Some(block) = self
+            .state
+            .lock_async(|s| Box::pin(s.chain.archival_state().get_block(block)))
+            .await?
+        {
+            // .block_index_db.get(
+            //     crate::state::database::BlockIndexKey::Block(
+            //         // s.chain.blockchain_archival_state().archival_state.block_height_to_block_digests(block_height).await
+            //         b
+            //     )
+            // )
+            if let Some(tx_output) = self
+                .state
+                .lock_async(|s| s.wallet_state.wallet_db.sent_transactions().get(tx_ix))
+                .await
+                .tx_outputs
+                .get(utxo_ix)
+            {
+                // let utxo = tx.tx_outputs.utxos_iter().into_iter().skip(utxo_ix).next()?;//.map_err(todo![])?;
+                let utxo = tx_output.utxo();
+                let aocl = block.body().block_mmr_accumulator;
```

Comment:
Good news! Thanks to the recent changes to the wallet database (now on master), the field `aocl_leaf_index` now lives on `MonitoredUtxo`. I'm not sure about the exact sequence of dabase lookups to get the right `MonitoredUtxo` but I am sure it's a lot easier with this change than it was before.